src/Exchange/ExchangerExchange.php

```diff
@@ -13,35 +15,27 @@
 
 /**
  * Provides a way to get exchange rate from a third-party source and return a currency pair.
- *
- * @author Maksim (Ellrion) Platonov <ellrion11@gmail.com>
  */
 final class ExchangerExchange implements Exchange
 {
-    /**
-     * @var ExchangeRateProvider
-     */
-    private $exchanger;
+    private ExchangeRateProvider $exchanger;
 
     public function __construct(ExchangeRateProvider $exchanger)
     {
         $this->exchanger = $exchanger;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function quote(Currency $baseCurrency, Currency $counterCurrency)
+    public function quote(Currency $baseCurrency, Currency $counterCurrency): CurrencyPair
     {
         try {
             $query = new ExchangeRateQuery(
                 new ExchangerCurrencyPair($baseCurrency->getCode(), $counterCurrency->getCode())
             );
-            $rate = $this->exchanger->getExchangeRate($query);
-        } catch (ExchangerException $e) {
+            $rate  = $this->exchanger->getExchangeRate($query);
+        } catch (ExchangerException) {
             throw UnresolvableCurrencyPairException::createFromCurrencies($baseCurrency, $counterCurrency);
         }
 
-        return new CurrencyPair($baseCurrency, $counterCurrency, $rate->getValue());
+        return new CurrencyPair($baseCurrency, $counterCurrency, (string) $rate->getValue());
```

Comment:
What does `$rate->getValue()` return here? If it is, or might be, float to string, then I think we should consider using `Number::fromFloat` or use a similar `sprintf` call as `Number::fromFloat` is using.

Reply:
It returns `float` in this case: I don't think this is performance-sensitive though, since we're doing I/O against an external system anyway

Reply:
My comment was not related to performance, but related to the inaccurate `(string)$float` conversion. I believe `sprintf('%.14F', $float)` is a better way to handle this, see [this discussion we had earlier](https://github.com/moneyphp/money/issues/408).

Reply:
9496d0940a031249a42c03bfab0ac79826ed7a22